### DISCUSSION TITLE
Assembly descriptions

### DISCUSSION THREAD

**Discussion by nathanweeks** [GMOD/jbrowse-components#3570] [Ideas] [answered]
Tracks currently support an optional description (e.g., `jbrowse add-track my.bam -d 'My description' ...`). It would be helpful if assemblies supported optional descriptions as well, especially for JBrowse instances that host many assemblies, to store a more verbose description of the assembly (including embedded links to external resources) than can be realistically captured in/derived from the assembly name or an alias. 

I'm not sure where an optional assembly description would best go in the JBrowse UI for aesthetics and ease of implementation, but it would be helpful to have accessible from somewhere.

**Comment by cmdcolin** [comment, marked as answer]
one potential way of doing this is using the 'metadata' field on the reference sequence track

Example

https://github.com/GMOD/jbrowse-components/blob/main/test_data/volvox/config.json#L9-L11

another is storing the extended description in a 'fasta metadata file'. this was something that was a project of a user https://github.com/GMOD/jbrowse-components/blob/main/test_data/config.json#L16-L19 https://github.com/GMOD/jbrowse-components/pull/2968

potentially more types of tracks should have this type of "arbitrary external metadata file" but not available yet


I don't know off hand how to set the metadata and metdataLocation via the CLI tool, but you can possibly hand edit it into the JSON or write a small script that adds it to the JSON

**Reply by cmdcolin** [reply]
note: the "About track" on the Reference sequence track will show these metadata fields in the GUI

**Reply by nathanweeks** [reply]
@cmdcolin: Thanks, both metadata and metadataLocation appear to be good solutions. It would be convenient if they could be set via the JBrowse CLI. In the meantime, my initial go at updating an existing config.json (using [jq](https://github.com/stedolan/jq)) to add a metadata.description for a given assembly name:
```
jq --arg name "${name}" \
   --arg description "${description}" \
   '(.assemblies[] | select(.name == $name).sequence.metadata.description)=$description' < config.json > config.json.jq
mv config.json.jq config.json
```
